### PR TITLE
Added `extend`/`extend_unchecked` for `MutableBinaryArray`

### PR DESCRIPTION
**Pull request by VasanthakumarV**
This PR adds `TrustedLen`-based extend and extend_unchecked methods for MutableBinaryArray.

### CODE REVIEW

File: src/array/binary/mutable.rs [hunk 1]
```diff
@@ -261,6 +261,77 @@ impl<O: Offset> MutableBinaryArray<O> {
...
+    }
+
+    /// Extends the [`MutableBinaryArray`] from an iterator of [`TrustedLen`]
+    #[inline]
+    pub fn extend_trusted_len<I, P>(&mut self, iterator: I)
+    where
+        P: AsRef<[u8]>,
+        I: TrustedLen<Item = Option<P>>,
+    {
+        // Safety: The iterator must be `TrustedLen`
```
Comment (jorgecarleitao) [inline src/array/binary/mutable.rs:304]:
```suggestion
        // Safety: The iterator is `TrustedLen`
```

File: src/array/binary/mutable.rs [hunk 2]
```diff
@@ -261,6 +261,77 @@ impl<O: Offset> MutableBinaryArray<O> {
...
 
+    /// Extends the [`MutableBinaryArray`] from an iterator of trusted length.
+    /// This differs from `extend_trusted_len` which accepts iterator of optional values.
+    #[inline]
+    pub fn extend_trusted_len_values<I, P>(&mut self, iterator: I)
+    where
+        P: AsRef<[u8]>,
+        I: TrustedLen<Item = P>,
+    {
+        // Safety: The iterator must be `TrustedLen`
```
Comment (jorgecarleitao) [inline src/array/binary/mutable.rs:272]:
```suggestion
        // Safety: The iterator is `TrustedLen`
```

**Review by jorgecarleitao — APPROVED** [review]
Looks amazing, as usual. Two minor nits, good to go!

File: src/array/binary/mutable.rs (hunk 1 above)
Comment (VasanthakumarV) [reply src/array/binary/mutable.rs]: Rewording makes sense, made the change

File: src/array/binary/mutable.rs (hunk 2 above)
Comment (VasanthakumarV) [reply src/array/binary/mutable.rs]: Made the change

**Pull request merged**